lib/std/zig/system.zig

```diff
@@ -287,38 +287,7 @@ pub const NativeTargetInfo = struct {
                     os.version_range.windows.max = @intToEnum(Target.Os.WindowsVersion, version);
                     os.version_range.windows.min = @intToEnum(Target.Os.WindowsVersion, version);
                 },
-                .macos => {
-                    var scbuf: [32]u8 = undefined;
-                    var size: usize = undefined;
-
-                    // The osproductversion sysctl was introduced first with 10.13.4 High Sierra.
-                    const key_osproductversion = "kern.osproductversion"; // eg. "10.15.4"
-                    size = scbuf.len;
-                    if (std.os.sysctlbynameZ(key_osproductversion, &scbuf, &size, null, 0)) |_| {
-                        const string_version = scbuf[0 .. size - 1];
-                        if (std.builtin.Version.parse(string_version)) |ver| {
-                            os.version_range.semver.min = ver;
-                            os.version_range.semver.max = ver;
-                        } else |err| switch (err) {
-                            error.Overflow => {},
-                            error.InvalidCharacter => {},
-                            error.InvalidVersion => {},
-                        }
-                    } else |err| switch (err) {
-                        error.UnknownName => {
-                            const key_osversion = "kern.osversion"; // eg. "19E287"
-                            size = scbuf.len;
-                            std.os.sysctlbynameZ(key_osversion, &scbuf, &size, null, 0) catch {
-                                @panic("unable to detect macOS version: " ++ key_osversion);
-                            };
-                            if (macos.version_from_build(scbuf[0 .. size - 1])) |ver| {
-                                os.version_range.semver.min = ver;
-                                os.version_range.semver.max = ver;
-                            } else |_| {}
-                        },
-                        else => @panic("unable to detect macOS version: " ++ key_osproductversion),
-                    }
-                },
+                .macos => macos.detect(&os) catch {}, // valid to ignore any error and keep os defaults
```

Comment:
Doesn't this mean that if the user is running with less permissions, then they silently get different behaviour?

Shouldn't we at least `log.warn` that we had an error and are falling back?

Reply:
on error the .linux prong goes with defaults. Prior to this PR .macos prong would `@panic` if what we think is invariant behaviour on macOS doesn't hold true. The analogous thing to do in this PR would be to `@panic` on say the system .plist file being unreadable which means someone has a corrupt macOS. I don't think there's too much value in that.

Reply:
> the system .plist file being unreadable which means someone has a corrupt macOS

or they're running in a build chroot without it
or they hit their `ulimit`
or they use one of the security products that can block access to files
etc....

Reply:
.linux error handling:
- `std.builtin.Version.parse` error → default

.macos error handling:
1. file error → **choose one** { default, default+log, panic }
2. .plist parser error → **choose one** { default, default+log, panic }
3. `std.builtin.Version.parse` error → default

are we ok with macos { 1, 2 } → default+log ?

Reply:
In general I'm not a fan of using `warn`. If it's a problem, we should deal with it appropriately, and probably fail. If it's not a problem, why do we need to warn about it?

It looks like with the .linux prong, I made the call that a failure to parse the linux version name would be solved by using the default version range. Logic here is that linux is a diversely configured kernel that perhaps could have strange distros with non-standard uname formats. Or perhaps there is some seccomp stuff going on that is making uname not available. In such cases the default linux version range would be OK.

For macOS, the system is more uniform. I think we should expect version detection to work and report failure if it does not.